lib/tenstorrent/bh_arc/init.c

```diff
@@ -427,6 +473,21 @@ static int InitHW(void)
 		}
 	}
 
+	SetPostCode(POST_CODE_SRC_CMFW, POST_CODE_ARC_INIT_STEPE);
+	/* Check GDDR training status. */
+	if (!IS_ENABLED(CONFIG_TT_SMC_RECOVERY)) {
+		k_timepoint_t timeout = sys_timepoint_calc(K_MSEC(MRISC_INIT_TIMEOUT));
+
+		for (uint8_t gddr_inst = 0; gddr_inst < NUM_GDDR; gddr_inst++) {
+			if (IS_BIT_SET(GetDramMask(), gddr_inst)) {
+				if (CheckGddrTraining(gddr_inst, timeout)) {
+					LOG_ERR("GDDR instance %d failed training.\n", gddr_inst);
```

Comment:
Let's include the return value of `CheckGddrTraining` in the log here so it is obvious if the failure was a timeout or not